app/middleware/selectChartTypeMiddleware.js

```diff
@@ -0,0 +1,42 @@
+import {
+  RECEIVE_CHART_DATA,
+  RECEIVE_CHART_DATA_INIT,
+  RECEIVE_CHART_OPTIONS,
+  RECEIVE_CHART_OPTIONS_INIT,
+  SELECT_CHART_TYPE,
+} from '../constants';
+import actionTrigger from '../actions';
+import { getChartTypeDefaultOpts } from '../constants/chartTypes';
+
+export default function({ getState }) {
+  return (dispatch) => (action) => {
```

Comment:
I'm not a huge fan of this as the second function is so many lines... I'd be okay with `return (x) => (y) => x + y`

But it's valid, and doesn't break linting standards. I would like you to wrap the returned function with parans although (see https://github.com/airbnb/javascript#arrows--paren-wrap), for readability

`return (dispatch) => ((action) => {});`